### PR TITLE
[gen_l10n] Create pubspec.yaml in ".dart_tool/flutter_gen" if it does not already exist

### LINKED ISSUE #2835: [build_runner 1.10.3] Bad state: Unable to generate package graph
`build_runner: ^1.10.3`
` flutter packages pub run build_runner build`:
```
Unhandled exception:
Bad state: Unable to generate package graph, no `/home/aliyazdi75/Documents/AndroidStudioProjects/koja_beram_app/.dart_tool/flutter_gen/pubspec.yaml` found.
#0      _pubspecForPath (package:build_runner_core/src/package_graph/package_graph.dart:235:5)
#1      _parsePackageDependencies (package:build_runner_core/src/package_graph/package_graph.dart:208:21)
#2      PackageGraph.forPath (package:build_runner_core/src/package_graph/package_graph.dart:97:33)
<asynchronous suspension>
#3      PackageGraph.forThisPackage (package:build_runner_core/src/package_graph/package_graph.dart:110:20)
#4      main (file:///home/aliyazdi75/snap/flutter/common/flutter/.pub-cache/hosted/pub.dartlang.org/build_runner-1.10.3/bin/build_runner.dart:27:49)
#5      _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:299:32)
#6      _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:168:12)
pub finished with exit code 255
```
`flutter doctor -v`:
```
[✓] Flutter (Channel master, 1.22.0-10.0.pre.317, on Linux, locale en_US.UTF-8)
    • Flutter version 1.22.0-10.0.pre.317 at /home/aliyazdi75/snap/flutter/common/flutter
    • Framework revision 10febea877 (7 hours ago), 2020-09-23 07:22:10 +0200
    • Engine revision 57ee3d4b32
    • Dart version 2.10.0 (build 2.10.0-156.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /home/aliyazdi75/Android/Sdk
    • Platform android-30, build-tools 29.0.3
    • Java binary at: /home/aliyazdi75/Application/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • clang version 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
    • cmake version 3.10.2
    • ninja version 1.8.2
    • pkg-config version 0.29.1

[✓] Android Studio (version 4.0)
    • Android Studio at /home/aliyazdi75/Application/android-studio
    • Flutter plugin version 49.0.2
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] Connected device (3 available)
    • Linux (desktop)  • linux      • linux-x64      • Linux
    • Web Server (web) • web-server • web-javascript • Flutter Tools
    • Chrome (web)     • chrome     • web-javascript • Google Chrome 85.0.4183.12
```

### PR DESCRIPTION
**Pull request by shihaohong**
Fixes https://github.com/dart-lang/build/issues/2835

## Tests

I added the following tests:
- A test to ensure that a valid pubspec.yaml file is created
- A test to ensure that existing pubspec.yaml file is not overwritten

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter_tools/lib/src/localizations/gen_l10n_templates.dart [hunk 1]
```diff
@@ -2,6 +2,10 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+const String emptyPubspecTemplate = '''name: synthetic_package
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/localizations/gen_l10n_templates.dart:5]: I think this also needs to have an SDK constraint too? I'm not sure if build will prefer the one in the package_config.

**Review by jonahwilliams — APPROVED** [review]
LGTM with question

File: packages/flutter_tools/lib/src/localizations/gen_l10n_templates.dart (hunk 1 above)
Comment (shihaohong) [reply packages/flutter_tools/lib/src/localizations/gen_l10n_templates.dart:5]: The `build_runner` exception no longer appears with a pubspec that does not specify an SDK constraint. Could you clarify what you mean by build preferring the one in package_config over providing one here?

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/localizations/gen_l10n_templates.dart:5]: If it doesn't happen at all, then I would leave this as minimal as possible and with a comment what generated it

**Comment by kuhnroyal** [comment]
This change looks small, can this be picked into a 1.22 hotfix?

**Comment by shihaohong** [comment]
Since there's a workaround for this and it's not a breaking regression, I don't think it fits into the description of a hotfix: https://github.com/flutter/flutter/wiki/Hotfixes-to-the-Stable-Channel

**Pull request merged**